src/main/java/com/acoustic/SpringPolandSalaryCalculator/controller/SalaryCalculatorController.java

```diff
@@ -0,0 +1,124 @@
+import javax.validation.constraints.NotNull;
+
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.acoustic.SpringPolandSalaryCalculator.calculatorservice.SalaryCalculatorService;
+import com.acoustic.SpringPolandSalaryCalculator.entity.DataSalaryCalculator;
+import com.acoustic.SpringPolandSalaryCalculator.jobcategories.JobCategoriesConfigurationProperties;
+import com.acoustic.SpringPolandSalaryCalculator.service.DataSalaryCalculatorRepository;
+import lombok.RequiredArgsConstructor;
+
+
+@RestController
+@RequiredArgsConstructor
+@Validated
+public class SalaryCalculatorController {
+
+    public static final int MINIMUM_GROSS = 2000;
+    private final DataSalaryCalculatorRepository dataSalaryCalculatorRepository;
+    private final List<SalaryCalculatorService> salaryCalculatorServices;
+    private final JobCategoriesConfigurationProperties jobCategoriesConfigurationProperties;
+
+
+    @GetMapping("/getJobTitles/{departmentName}")
+    public List<String> getJobTitles(
+            @PathVariable
+            String departmentName) {
+        return this.jobCategoriesConfigurationProperties.getJobDepartmentAndTitles().get(departmentName);
+    }
+
+    @GetMapping("/getJobDepartments")
+    public Set<String> getDepartmentName() {
+        return this.jobCategoriesConfigurationProperties.getJobDepartmentAndTitles().keySet();
+    }
+
+
+    @PostMapping("/calculator/{grossMonthlySalary}")
+    public Map<String, BigDecimal> calculateSalary(
+            @PathVariable
+            @Min(value = MINIMUM_GROSS,
+                    message = "Must be Greater than or equal to 2000.00") @NotNull BigDecimal grossMonthlySalary,
+            @RequestParam(required = false)
+            String departmentName,
+            @RequestParam(required = false)
+            Integer jobTitleId) {
+
+        var response = responseWithoutStatistics(grossMonthlySalary);
+        if (departmentName == null || jobTitleId == null) {
+            return response;
+        }
+        return getAverage(grossMonthlySalary, departmentName, jobTitleId, response);
+
+    }
+
+    private Map<String, BigDecimal> responseWithoutStatistics(BigDecimal grossMonthlySalary) {
+        Map<String, BigDecimal> response = new LinkedHashMap<>();
+        salaryCalculatorServices.sort(Comparator.comparingInt(SalaryCalculatorService::getCalculationOrder));
+        BigDecimal grossMonthlySalaryMinusTaxes = grossMonthlySalary;
+        BigDecimal tax = null;
+        final int taxesOrder = 3;
+        final int netOrder = 4;
+        for (var service : salaryCalculatorServices) {
+            if (service.getCalculationOrder() > 5) {
```

Comment:
Why do we have some magic threshold? Cannot we encode this information in these services to be more flexible?

Reply:
I have created calculation-order.properties file, I have removed final int taxesOrder = 3 and
        final int netOrder = 4 the properties file values/
Example: if (service.getCalculationOrder() > this.serviceCalculationOrder.getAnnualNetServiceOrder())